cohortextractor2/dsl.py

```diff
@@ -161,9 +169,109 @@ class PatientSeries:
     def __init__(self, value: Value):
         self.value = value
 
+    @staticmethod
+    def _get_other(other: Expression) -> Comparator | Value:
+        if isinstance(other, PatientSeries):
+            return other.value
+        return other
 
-Expression = Union[PatientSeries, bool, int, str]
+    def __gt__(self, other: Expression) -> Comparator:
+        return self.value > self._get_other(other)
 
+    def __ge__(self, other: Expression) -> Comparator:
+        return self.value >= self._get_other(other)
 
-def categorise(mapping: dict[str, Expression]) -> PatientSeries:
-    """Represents a switch statement."""
+    def __lt__(self, other: Expression) -> Comparator:
+        return self.value < self._get_other(other)
+
+    def __le__(self, other: Expression) -> Comparator:
+        return self.value <= self._get_other(other)
+
+    def __eq__(self, other: Expression) -> Comparator:  # type: ignore[override]
+        return self.value == self._get_other(other)
+
+    def __ne__(self, other: Expression) -> Comparator:  # type: ignore[override]
+        return self.value != self._get_other(other)
+
+    def __and__(self, other: Expression) -> Comparator:
+        return self.value & self._get_other(other)
+
+    def __or__(self, other: Expression) -> Comparator:
+        return self.value | self._get_other(other)
+
+    def __invert__(self) -> Comparator:
+        return ~self.value
+
+    def __repr__(self) -> str:
+        return f"PatientSeries(value={self.value})"
+
+
+Expression = Union[PatientSeries, Comparator]
+
+
+def categorise(mapping: dict[str, Expression], default: bool | None) -> PatientSeries:
```

Comment:
The possibility of passing `None` for `default` isn't exercised in the tests. Do we mean to support it? If so, shouldn't we provide a default value for the argument of `None`, rather than expecting it to be passed explicitly?

Reply:
Yes, that's an error,  it should default to `None`